### ISSUE TITLE
provide Python builds for linux-aarch64-musl

### ISSUE THREAD

**Issue by mikolysz** [astral-sh/uv#6890]
linux-aarch64-musl is the platform one gets when using an Alpine-based Docker image and developing on Apple Silicon, which is a pretty common setup at this point. This set up currently (as of uv 0.4.1) doesn't support installing Python via uv, with the following error:

```
Searching for Python installations
error: No download found for request: cpython-any-linux-aarch64-musl
```

For other users encountering this issue, installing Python via `app add python3` seems to work for now, although Alpine's packages tend to be fairly minimal, so more dependencies might be needed for larger projects.

**Comment by charliermarsh** [comment]
This is known but can't find an existing issue to link to -- will leave it up for now.

**Comment by charliermarsh** [comment]
Related to https://github.com/astral-sh/uv/pull/6643.

**Comment by zanieb** [comment]
Yeah more context in

- https://github.com/astral-sh/uv/issues/4242
- https://github.com/astral-sh/uv/issues/6392

**Comment by SamuelMarks** [comment]
What's the current progress here? - Need a hand?

**Comment by zanieb** [comment]
If you're interested in lending a hand it's definitely welcome! I haven't started working on this part of `python-build-standalone` yet.

Briefly, it looks like there were some previous discussions:

- https://github.com/astral-sh/python-build-standalone/issues/86
- https://github.com/astral-sh/python-build-standalone/issues/260

https://github.com/astral-sh/python-build-standalone/issues/86 seems like the best place to discuss implementation details, if necessary.

**Comment by SamuelMarks** [comment]
Ok, just took a look: https://github.com/astral-sh/python-build-standalone/releases/tag/20241219 doesn't have a build for my architecture + Linux + musl.

I'll get my other laptop back in a couple of days will try then

Happy New Year!

**Comment by mwarkentin** [comment]
I ran into this when attempting to use `uv` inside of an Alpine docker container (which is the first "derived image" mentioned on https://docs.astral.sh/uv/guides/integration/docker/#available-images. 

My use case: working on a docker container that contains a grab bag of scripts that our team has built over time, and hoping to use `uv` to manage python versions and dependencies which may be different for each script, using the inline `/// script` definitions.

Will switch over to a Debian image for now instead.

**Comment by zanieb** [comment]
We support these as of 0.6.6 — https://github.com/astral-sh/uv/pull/12121

**Comment by mio-19** [comment]
Would it be difficult to add aarch64-musl?

**Comment by zanieb** [comment]
Yeah, it is. See https://github.com/astral-sh/python-build-standalone/pull/569#issue-2933229417

**Comment by matt-dies-tenet3** [comment]
Looks like the [latest release in `astral-sh/python-build-standalone`](https://github.com/astral-sh/python-build-standalone/releases/tag/20250828) includes "dynamically linked aarch64 musl builds"! 🥳

**Comment by zanieb** [comment]
Yep, thanks!

**Comment by kopf** [comment]
> Looks like the [latest release in `astral-sh/python-build-standalone`](https://github.com/astral-sh/python-build-standalone/releases/tag/20250828) includes "dynamically linked aarch64 musl builds"! 🥳

Does this (and the fact that this issue is closed) mean that the following shouldn't happen?

```
bash-5.2$ uv self version
uv 0.9.18
bash-5.2$ uv python install 3.12
error: uv does not yet provide musl Python distributions on aarch64.
bash-5.2$ uname -a
Linux OpenWrt 6.6.73 #0 SMP Mon Feb  3 23:09:37 2025 armv7l GNU/Linux
bash-5.2$
```

Or is there another blocker to what I'm trying to do? (Install python via uv on openwrt running on a linksys wrt3200acm)

**Comment by zanieb** [comment]
Hm...

```
❯ docker run --rm -it ghcr.io/astral-sh/uv:alpine sh -c "uv python install 3.12"
Installed Python 3.12.12 in 2.44s
 + cpython-3.12.12-linux-aarch64-musl (python3.12)
```

armv7l is 32-bit, right? We don't build Python for that target.

**Comment by classabbyamp** [comment]
yes, armv7l != aarch64

**Comment by kopf** [comment]
@zanieb, @classabbyamp, aha, okay. Sorry, my fault - the error message made me jump to google and write here before pausing to give it some thought. 

I take it there's no armv7l support planned in the future?

**Comment by zanieb** [comment]
Not that I'm aware of. Where are you running that?

We should fix the error message though.

**Comment by kopf** [comment]
> Not that I'm aware of. Where are you running that?

OK, understandable. I'm running it on my router, a linksys wrt3200acm that I've installed openwrt on. The exotic setup is also a big part of the reason I'm writing here - I'd love to have `uv` managing my python installation(s) instead of leaving it to the system package manager which honestly isn't up to the task.

**Comment by zanieb** [comment]
It looks like it's being tracked in https://github.com/astral-sh/python-build-standalone/issues/229

I'm not sure how hard it'd be to support. I presume non-trivial though.
